File: ros2_project/src/3-1introduce_launch/test_launch_py/test_launch_py/test_node1.py
import rclpy
from rclpy.node import Node

class TestNode1(Node):
    def __init__(self):
        super().__init__("TestNode1")

        self.get_logger().info("你好世界1")

def main(args=None):
    rclpy.init(args=args)   # 初始化rclpy
    node = TestNode1()      # 新建一个节点
    rclpy.spin(node)        # 保持节点运行，检测是否收到退出指令(Ctrl+C)
    rclpy.shutdown()        # 当接收到退出指令时，退出spin函数，执行shutdown函数来关闭rclpy

if __name__=="__main__":
    main()

File: ros2_project/src/3-1introduce_launch/test_launch_cpp/src/test_node1.cpp
#include <rclcpp/rclcpp.hpp>

class TestNode1 : public rclcpp::Node
{
public:
    TestNode1():rclcpp::Node("TestNode1")
    {
        RCLCPP_INFO(this->get_logger(),"你好世界1");
    }
};

int main(int argc,char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<TestNode1>();
    rclcpp::spin(node);
    rclcpp::shutdown();
}